pkg/prometheus/promcfg.go

```diff
@@ -1120,7 +1120,7 @@ func (cg *ConfigGenerator) generateServiceMonitorConfig(
 		role = kubernetesSDRoleEndpointSlice
 	}
 
-	cfg = append(cfg, cg.generateK8SSDConfig(m.Spec.NamespaceSelector, m.Namespace, apiserverConfig, store, role, nil))
+	cfg = append(cfg, cg.generateK8SSDConfig(m.Spec.NamespaceSelector, m.Namespace, apiserverConfig, store, role, m.Spec.AttachMetadata))
```

Comment:
I wonder do we need to version check here against `v2.37.0`

https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go#L1506-L1510 appears to take care of it generically but I am wondering if we need specific checks per CRD

Reply:
Good point! I think it might make sense, better be safe than sorry

Reply:
Should be done 👍